CMakeLists.txt

```diff
@@ -201,6 +211,14 @@ add_dependencies(psi4_core ambit_external
                            libint_external
                            pcmsolver_external
                            pybind11_external)
+if(ENABLE_psi4pasture)
+  ExternalProject_Add(psi4pasture
+    GIT_REPOSITORY https://github.com/amjames2/psi4pasture
```

Comment:
because of the superbuild, externals is odd in that for each of them, either a real or dummy library is always built anyway. i think make an internals directory and `add_subdirectory(internals)` here. as you point out, this needs to happen at a different point in the main cmake processing, and i think we do want these hidden away in a subdir.

i think you've convinced me on using them as plugins instead of link in libraries. according to practice so far, they wouldn't even be included in the psi4 cmake build sys. instead you'd build on your own and make avaailble through pythonpath at runtime. but i think it is nice to be able to order them up though the main build. in which case maybe external --> extlibraries and internal --> extplugins. but that's for general discussion.